bears/vcs/git/GitCommitBear.py

```diff
@@ -102,6 +105,70 @@ def get_host_from_remotes():
             netloc = urlparse(url)[1]
         return netloc.split('.')[0]
 
+    @staticmethod
+    def get_owner_from_url(url):
+        return urlparse(url)[2].split('/')[1]
+
+    @staticmethod
+    def get_repository_from_url(url):
+        return urlparse(url)[2].split('/')[2]
+
+    @staticmethod
+    def get_owner_from_remote(remote):
```

Comment:
Surely there are libraries that do all of this? :P

We've been talking about libraries for this in the community and coala-bears rooms.

Reply:
Yes, I am very well aware of this issue. I didn't push new changes since there are no means of checking GitLab repo without auth token as we have discussed [here](https://gitlab.com/gitmate/open-source/IGitt/issues/114). I think this issue should be blocked until environment variables or some other mechanism to read user-specific settings gets introduced in coala.